### PR TITLE
refactor(popover-container): convert to ts - FE-4819

### LINKED ISSUE #5152: [PopoverContainer ] Differences between PopoverContainer component and types
### Current behaviour

In PopoverContainer  props `renderOpenComponent` and `renderCloseComponent` expected `PopoverContainerProps.renderCloseComponent?: React.ReactNode | Node` as prop type.  With caused type error and forced to cast to any. 

### Expected behaviour

To be consisted with [story book description](https://carbon.sage.com/?path=/docs/popover-container--default-story) and [implementation](https://github.com/Sage/carbon/blob/c8d70201dc19e5bb27395686e66ddc6b81fc3379/src/components/popover-container/popover-container.component.js#L69) this should be a render props function. 

### CodeSandbox or Storybook URL

https://codesandbox.io/s/hungry-cloud-3opd74?file=/src/index.tsx

### JIRA Ticket (Sage Only)

SBS-33979

### Suggested Solution

Based on [implementatnion](https://github.com/Sage/carbon/blob/c8d70201dc19e5bb27395686e66ddc6b81fc3379/src/components/popover-container/popover-container.component.js#L69) create correct interface in [types file](https://github.com/Sage/carbon/blob/c8d70201dc19e5bb27395686e66ddc6b81fc3379/src/components/popover-container/popover-container.d.ts#L6), same to `renderCloseComponentProps` prop. 

### Carbon Version

106.3.2

### Design Tokens Version

_No response_

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

Windows

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour

This PR converts PopoverContainer to TypeScript
This PR also fixes FE-5172 and closes #5152 
### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

PopoverContainer regression

### CODE REVIEW

File: src/components/popover-container/popover-container.spec.tsx [hunk 1]
```diff
@@ -392,17 +417,22 @@ describe("PopoverContainer", () => {
     });
 
     describe("and custom component is provided as a closing button", () => {
-      const MyCloseButton = forwardRef((props, ref) => (
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const MyCloseButton = forwardRef<HTMLButtonElement, any>((props, ref) => (
```
Comment (edleeks87) [inline src/components/popover-container/popover-container.spec.tsx:421]: `RenderCloseComponentArgs` instead of `any`?

**Review by edleeks87 — COMMENTED** [review]
FYI you've forgotten to remove the `popover-container.d.ts` file

**Review by edleeks87 — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

File: src/components/popover-container/popover-container.component.tsx [hunk 2]
```diff
@@ -14,33 +14,123 @@ import Icon from "../icon";
...
+  >
+    <Icon type="close" />
+  </PopoverContainerCloseIcon>
 );
 
-const PopoverContainer = ({
+export interface PopoverContainerProps extends PaddingProps {
+  /** A function that will render the open component
+   *
+   * `({tabIndex, isOpen, data-element, onClick, ref, aria-label}) => ()`
```
Comment (grabkowski) [inline src/components/popover-container/popover-container.component.tsx:77]: do we still need that information here?

Comment (mkrds) [reply src/components/popover-container/popover-container.component.tsx:77]: Probably not, but it's actually more readable in storybook thanks to this comment.

**Review by grabkowski — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

File: src/components/popover-container/popover-container.component.tsx [hunk 3]
```diff
@@ -71,22 +161,24 @@ const PopoverContainer = ({
...
     onClick: handleCloseButtonClick,
     ref: closeButtonRef,
-    ariaLabel: closeButtonAriaLabel,
+    "aria-label": closeButtonAriaLabel,
   };
 
-  const handleClickAway = (e) => {
+  // TODO: Assign proper type after ClickAwayWrapper has been refactored
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const handleClickAway = (e: any) => {
```
Comment (edleeks87) [inline src/components/popover-container/popover-container.component.tsx:181]: See related comment above
```suggestion
  const handleClickAway = (e: CustomEvent) => {
```

File: src/components/popover-container/popover-container.component.tsx [hunk 4]
```diff
@@ -15,34 +15,124 @@ import createGuid from "../../__internal__/utils/helpers/guid";
...
+   */
+  renderCloseComponent?: (args: RenderCloseProps) => JSX.Element;
+  /** The content of the popover-container */
+  children?: React.ReactNode;
+  /** Sets rendering position of dialog */
+  position?: "left" | "right";
+  /** Sets the popover container dialog header name */
+  title?: string;
+  /** Callback fires when close icon clicked */
+  onClose?: (ev: React.MouseEvent<HTMLElement>) => void;
```
Comment (edleeks87) [inline src/components/popover-container/popover-container.component.tsx:95]: I could be wrong or suggesting something heretical here but setting the type like so would mean we don't need to use `any`
```suggestion
  onClose?: (ev: React.MouseEvent<HTMLElement> | CustomEvent) => void;
```

**Review by edleeks87 — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**